### PR TITLE
add a job handler to stream electorates for an election

### PR DESCRIPTION
**Pull request by theJohnnyBrown**
Also adds an optional flag to the election-create event so a client that uses this won't create duplicate electorates.

This is part of https://www.pivotaltracker.com/n/projects/930636/stories/157370946, see also PR in election-works coming soon

### CODE REVIEW

**Comment by theJohnnyBrown** [comment]
@tie-rack fyi

File: src/electorate_works/handlers/election.clj [hunk 1]
```diff
@@ -14,16 +14,30 @@
 (defn-traced handle-create*
   [{election :v2}]
   (log/debug "Handling an election.create event:" (pr-str election))
-  (let [electorates (lookup/electorates-from-election election)]
-    (loop []
-      (when-let [electorate (async/<!! electorates)]
-        (e/create-and-broadcast-electorate (e/->id-map electorate)
-                                           {:election election})
-        (recur)))))
+  (when (:create-electorates election true)
```
Comment (ddellacosta) [inline src/electorate_works/handlers/election.clj]: If this always evaluates to true, then you don't need it in here do you?

File: src/electorate_works/handlers/election.clj [hunk 2]
```diff
@@ -14,16 +14,30 @@
...
+     (loop []
+       (when-let [electorate (async/<!! electorates)]
+         (e/create-and-broadcast-electorate (e/->id-map electorate)
+                                            {:election election})
+         (recur))))))
 
 (def handle-create
   (handlers/handle-errors handle-create*))
 
+(defn-traced stream-electorates-for-election
```
Comment (mrichards42) [inline src/electorate_works/handlers/election.clj]: To me, `stream-electorates-for-election` sounds like a function used for searching. What do you think about `create-electorates-job`, or something like that?

File: src/electorate_works/handlers/election.clj (hunk 1 above)
Comment (mrichards42) [reply src/electorate_works/handlers/election.clj]: If election includes `{:create-electorates false}` that would evaluate to false.

Comment (mrichards42) [reply src/electorate_works/handlers/election.clj]: If election includes `{:create-electorates false}` that would evaluate to false.

Comment (ddellacosta) [reply src/electorate_works/handlers/election.clj]: Oh I see, I had it backwards. Thanks @mrichards42 , nevermind!

**Comment by theJohnnyBrown** [comment]
Regarding the `:create-electorates false` thing, the intention is that when a client (election-works) uses the job to create electorates for an election as part of an election create, it would then dispatch the election.create event with that flag so the electorates would not be created again. 

The purpose of this is for an election import to create an election and then "stream" all the electorates for that election before creating another election, so that we aren't creating electorates for several large elections at once and overloading the queues.

File: src/electorate_works/handlers/election.clj (hunk 2 above)
Comment (theJohnnyBrown) [reply src/electorate_works/handlers/election.clj]: I'm ok with renaming that 👍

**Pull request merged**